gdal/swig/include/cpl.i

```diff
@@ -520,6 +521,27 @@ void wrapper_VSIFileFromMemBuffer( const char* utf8_path, int nBytes, const GByt
 #endif
 #endif
 
+#if defined(SWIGJAVA)
+%inline %{
+void wrapper_VSIFileFromMemBuffer( const char* utf8_path, void* nioBuffer, long nioBufferSize) {
+    VSIFCloseL(VSIFileFromMemBuffer(utf8_path, (GByte*) nioBuffer, nioBufferSize, FALSE));
+}
+%}
+#endif
+
+#if defined(SWIGJAVA)
+%apply (jobject outBuffer) {jobject};
+%inline %{
+jobject wrapper_VSIGetMemFileBuffer(JNIEnv *jenv, const char* utf8_path) {
```

Comment:
hum I don't think that playing directly with JNIEnv* here is the most idiomatic way of proceeding. Ideally we should minimize the amount of language specificities here. A cleaner way would probably to have something like
void wrapper_VSIGetMemFileBuffer(const char* utf8_path, void** outBuffer, size_t* outBufferSize) and a typemap for (void** outBuffer, size_t* outBufferSize).